File: Java/src/main/java/swim/Data/Repositories/UserRepositories/UserRepository.java
package swim.Data.Repositories.UserRepositories;

import swim.Data.Repositories.DatabaseRepository;
import swim.Domain.Models.User;

import java.util.ArrayList;
import java.util.List;

public class UserRepository extends DatabaseRepository<Integer, User> {

    public Boolean delete(Integer id)
    {
        logger.info("Deleting User: {}", id);
        String queryString = "Delete From Users Where id = ?";

        try{
            var statement = connection.Get().prepareStatement(queryString);
            statement.setInt(1, id);

            var result = statement.executeUpdate() > 0;

            if (result)
            {
                logger.info("Delete Successful");
            }
            else
            {
                logger.warn("Delete Failed");
            }

            return result;
        }
        catch(Exception e){
            throw new RuntimeException(e);
        }
    }

    public List<User> findAll()
    {
        logger.info("Searching all users");
        String queryString = "SELECT *  FROM  Users";

        try
        {
            var statement = connection.Get().prepareStatement(queryString);
            var users = new ArrayList<User>();

            var results = statement.executeQuery();
            while (results.next())
            {
                var id = results.getInt("id");
                var username = results.getString("username");
                var password = results.getString("password");

                var user = new User(id, username, password);

                users.add(user);
            }

            logger.info("Search Finished With {} results", users.size());
            return users;
        }
        catch (Exception e){
            throw new RuntimeException(e);
        }
    }

    public User findOne(Integer id)
    {
        logger.info("Searching User: {}", id);
        String queryString = "SELECT *  FROM  Users where id = ?";

        try
        {
            var statement = connection.Get().prepareStatement(queryString);
            statement.setInt(1, id);

            var results = statement.executeQuery();

            User user;
            if (results.next())
            {
                logger.info("User Found");

                var username = results.getString("username");
                var password = results.getString("password");

                user = new User(id, username, password);
            }
            else
            {
                logger.warn("User Not Found");
                user = null;
            }

            logger.info("Search finished");
            return user;
        } catch (Exception e) {
            throw new RuntimeException(e);
        }


    }

    public Boolean save(User entity)
    {
        logger.info("Saving User: {}", entity.getId());
        String queryString = "Insert Into Users(username, password) Values(?, ?)";

        try{
            var statement = connection.Get().prepareStatement(queryString);
            statement.setString(1, entity.getUsername());
            statement.setString(2, entity.getPassword());

            var result = statement.executeUpdate() > 0;

            if (result)
            {
                logger.info("Save Successful");
            }
            else
            {
                logger.warn("Save failed");
            }

            return result;
        }
        catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public Boolean update(User entity)
    {
        logger.info("Updating User: {}", entity.getId());
        String queryString = "Update Users set username = ?, password = ? Where id = ?";

        try{
            var statement = connection.Get().prepareStatement(queryString);
            statement.setString(1, entity.getUsername());
            statement.setString(2, entity.getPassword());
            statement.setInt(3, entity.getId());

            var result = statement.executeUpdate() > 0;

            if (result)
            {
                logger.info("Update Successful");
            }
            else
            {
                logger.warn("Update failed");
            }

            return result;
        }
        catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: CSharp/Swim.Data/Repositories/UserRepositories/UserRepository.cs
using log4net;
using log4net.Core;
using Swim.Domain.Models.UserModels;
using System.Data.SqlClient;

namespace Swim.Data.Repositories.UserRepositories
{
    public class UserRepository : DatabaseRepository, IUserRepository
    {
        public bool Delete(int id)
        {
            _logger.InfoFormat("Deleting User: {0}", id);
            string queryString = "Delete From Users Where id = @id";


            var command = new SqlCommand(queryString, _connection.Get());
            command.Parameters.AddWithValue("@id", id);

            var result = command.ExecuteNonQuery() > 0;

            if (result)
            {
                _logger.Info("Delete Successful");
            }
            else
            {
                _logger.Warn("Delete Failed");
            }

            return result;
        }

        public IEnumerable<User> FindAll()
        {
            _logger.Info("Searching all users");
            string queryString = "SELECT *  FROM  Users";

            var command = new SqlCommand(queryString, _connection.Get());

            SqlDataReader reader = command.ExecuteReader();
            var users = new List<User>();
            try
            {
                while (reader.Read())
                {
                    var id = (int)reader["id"];
                    var username = reader["username"].ToString();
                    var password = reader["password"].ToString();

                    User user = new()
                    {
                        Id = id,
                        Username = username,
                        Password = password
                    };

                    users.Add(user);
                }
            }
            finally
            {
                reader.Close();
            }

            _logger.InfoFormat("Search Finished With {0} results", users.Count);
            return users;
        }

        public User? FindBy(string username, string password)
        {
            _logger.Info("Searching user");
            string queryString = "SELECT *  FROM  Users Where username = @username and password = @password";

            var command = new SqlCommand(queryString, _connection.Get());
            command.Parameters.AddWithValue("@username", username);
            command.Parameters.AddWithValue("@password", password);

            SqlDataReader reader = command.ExecuteReader();
            User? user = null;
            try
            {
                if (reader.Read())
                {
                    _logger.Info("User Found");

                    var id = (int)reader["id"];
                    var dbUsername = reader["username"].ToString();
                    var dbPassword = reader["password"].ToString();

                    user = new()
                    {
                        Id = id,
                        Username = dbUsername,
                        Password = dbPassword
                    };
                }
                else
                {
                    _logger.Warn("User Not Found");
                }
            }
            finally
            {
                reader.Close();
            }

            _logger.InfoFormat("Search Finished");
            return user;
        }

        public User? FindOne(int id)
        {
            _logger.InfoFormat("Searching User: {0}", id);
            string queryString = "SELECT *  FROM  Users where id = @id";

            var command = new SqlCommand(queryString, _connection.Get());
            command.Parameters.AddWithValue("@id", id);

            SqlDataReader reader = command.ExecuteReader();
            User? user = null;
            try
            {
                if (reader.Read())
                {
                    _logger.Info("User Found");

                    var username = reader["username"].ToString();
                    var password = reader["password"].ToString();

                    user = new()
                    {
                        Id = id,
                        Username = username,
                        Password = password
                    };                    
                }
                else
                {
                    _logger.Warn("User Not Found");
                }
            }
            finally
            {
                reader.Close();
            }

            _logger.Info("Search finished");
            return user;
        }

        public bool Save(User entity)
        {
            _logger.InfoFormat("Saving User: {0}", entity.Id);
            string queryString = "Insert Into Users(username, password) Values(@username, @password)";

            var command = new SqlCommand(queryString, _connection.Get());
            command.Parameters.AddWithValue("@username", entity.Username);
            command.Parameters.AddWithValue("@password", entity.Password);

            var result = command.ExecuteNonQuery() > 0;

            if (result)
            {
                _logger.Info("Save Successful");
            }
            else
            {
                _logger.Warn("Save failed");
            }

            return result;
        }

        public bool Update(User entity)
        {
            _logger.InfoFormat("Updating User: {0}", entity.Id);
            string queryString = "Update Users set username = @username, password = @password Where id = @id";


            var command = new SqlCommand(queryString, _connection.Get());
            command.Parameters.AddWithValue("@username", entity.Username);
            command.Parameters.AddWithValue("@password", entity.Password);
            command.Parameters.AddWithValue("@id", entity.Id);

            var result = command.ExecuteNonQuery() > 0;

            if (result)
            {
                _logger.Info("Update Successful");
            }
            else
            {
                _logger.Warn("Update failed");
            }

            return result;
        }
    }
}
